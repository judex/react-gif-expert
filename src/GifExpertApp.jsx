import { useState } from 'react';
import { AddCategory, GifGrid } from './components';


export const GifExpertApp = () => {

  const [categories, setcategories ] = useState(['One Punch']);

  const onAddCategory = ( newCategory ) => {    
    if( categories.includes(newCategory) ) return;    
    setcategories([newCategory, ...categories ]);
    // setcategories(cat => [...cat, newCategory]); //otra forma
  }

  return (
    <>
      <section className='container mx-auto p-8'>
        <h1 className='text-3xl font-bold mb-4'>GifExpertApp</h1>
        <div className="flex justify-between items-center gap-2 w-full mb-4">
          <AddCategory 
            onNewCategory={ (value) => onAddCategory(value) }
            // setcategories={ setcategories }
          />
          {/* <button type="button" className='py-4 px-4 bg-blue-500 flex-grow text-white font-semibold rounded shadow-md hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-blue-400 focus:ring-opacity-75 whitespace-nowrap' onClick={ addCategory }>Add category</button> */}
        </div>
        <ul className='ms-4 mt-4 divide-y divide-gray-200 list-decimal'>
          {
            categories.map( category => (
              <GifGrid 
                key={ category } 
                category={ category }
              />
          ))}
        </ul>
      </section>
    </>
  )
}
