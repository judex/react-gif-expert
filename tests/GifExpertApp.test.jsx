import { fireEvent, render, screen } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp";

describe('Pruebas en <GifExpertApp />', () => { 

    test( 'debe de hacer match con el snapshot', () => {
        const { container } = render( <GifExpertApp /> );
        expect( container ).toMatchSnapshot();
    });

    test( 'debe de mostrar una categoria', () => {
        render( <GifExpertApp /> );
        expect( screen.getByText('One Punch') ).toBeTruthy();
    });

    test ('funcionalidad de AddCategory', () => {
        const onAddCategory = jest.fn();
        render( <GifExpertApp onAddCategory={ onAddCategory } /> );
        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');
        fireEvent.input(input, {target: {value: 'One Punch'}});
        fireEvent.submit(form);
        expect( input.value ).toBe('');
    })
})
    