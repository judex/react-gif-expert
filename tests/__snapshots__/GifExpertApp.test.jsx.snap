// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <section
    class="container mx-auto p-8"
  >
    <h1
      class="text-3xl font-bold mb-4"
    >
      GifExpertApp
    </h1>
    <div
      class="flex justify-between items-center gap-2 w-full mb-4"
    >
      <form
        aria-label="form"
        class="w-full"
      >
        <input
          class="w-full py-4 px-4 bg-white font-semibold rounded border-2 focus:border-blue-400"
          placeholder="Buscar Gif"
          type="text"
          value="One punch"
        />
      </form>
    </div>
    <ul
      class="ms-4 mt-4 divide-y divide-gray-200 list-decimal"
    >
      <li
        class="py-4"
      >
        <h3
          class="text-2xl font-bold mb-5"
        >
          One Punch
        </h3>
        <div
          class="text-sm font-bold mb-5"
        >
          Cargando...
        </div>
        <ol
          class="grid sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 xl:grid-cols-5 gap-6"
        />
      </li>
    </ul>
  </section>
</div>
`;
